sqlmesh/core/plan/definition.py

```diff
@@ -684,10 +699,11 @@ def _ensure_no_forward_only_revert(self) -> None:
 
     def _ensure_no_broken_references(self) -> None:
         for snapshot in self.context_diff.snapshots.values():
-            broken_references = set(self.context_diff.removed_snapshots) & snapshot.node.depends_on
+            broken_references = set(self.context_diff.removed_snapshots) & set(snapshot.parents)
```

Comment:
why was this changed? it seems reasonable but wondering your thoughts

Reply:
hmmmmmm, i think this is external model related, i think we'd need to talk and think this through. not sure if it's correct.

Reply:
Yeah I'm not sure about this either. It would be great if we had a test that could verify how this should behave.

What I can do is convert the removed snapshot ids to names (taken from the id) and then compare that against the names in snapshot parent ids.

Reply:
Updated to do what I described above

Reply:
Implemented the way described above